addons/web_hierarchy/static/src/hierarchy_model.js

```diff
@@ -670,29 +676,71 @@ export class HierarchyModel extends Model {
         config.context = "context" in params ? params.context : config.context;
         if ("domain" in params) {
             config.domain = params.domain;
-            if (!params.domain.length) {
-                if (config.context.hierarchy_res_id) {
-                    config.domain = [["id", "=", config.context.hierarchy_res_id]];
-                    delete config.context.hierarchy_res_id; // just needed for the first load.
-                } else {
-                    config.domain = this.defaultDomain;
+            if (this.isSearchDefaultOrEmpty() && config.context.hierarchy_res_id) {
+                config.domain = [["id", "=", config.context.hierarchy_res_id]];
+                const globalDomain = this.globalDomain;
+                if (globalDomain.length) {
+                    config.domain = Domain.and([config.domain, globalDomain]);
                 }
+                // Just needed for the first load.
+                delete config.context.hierarchy_res_id;
             }
         }
         return config;
     }
 
+    /**
+     * Evaluate if the current search query is the default one.
+     *
+     * @returns {boolean}
+     */
+    isSearchDefaultOrEmpty() {
+        if (!this.env.searchModel) {
+            return true;
+        }
+        const isDisabledOptionalSearchMenuType = (type) => {
+            return (
+                ["filter", "groupBy", "favorite"].includes(type) &&
+                !this.env.searchModel.searchMenuTypes.has(type)
+            );
+        };
+        const activeSearchItems = this.env.searchModel.getSearchItems(
+            (item) => item.isActive && !isDisabledOptionalSearchMenuType(item.type)
+        );
+        if (!activeSearchItems.length) {
+            return true;
+        }
+        const defaultSearchItems = this.env.searchModel.getSearchItems(
+            (item) =>
+                item.isDefault &&
+                item.type !== "favorite" &&
```

Comment:
Because they are not defined on the view arch.
My reasoning is that the view arch defines the widest scope possible for the view
and other filters and favorites can be anything and set by a user so we can not assume that the user did not make a specific selection of things that he wants to see, no matter if they are roots or not

In other words, I apply the "only roots" logic only in the most generic conditions, as soon as a user interacts with the view with a filter/favorite, I don't make any assumption on what the user wants to see

But I'm open for opinions^^ cc @xavierbol 

Maybe we'd want to add the logic on default favorites too

Reply:
Good question, I think for the moment the favorite and group by are disabled in the search view of hierarchy view, no?

In the first version of the hierarchy view, we had just kept the search on some fields to only search to find a specific record or the records with a specific parent. Now, since we allow the filter, we could indeed add the favorite filters since the filters are available and there is no longer any reason to disable it if I am not wrong. 🙂

Reply:
Also, since that task is for the freeze, I am not against to do it post freeze of course! 🙂